File: custom-event-bus/src/place-order.ts
import { APIGatewayProxyEvent, Handler, APIGatewayProxyResult } from 'aws-lambda';
import * as AWS from 'aws-sdk';
import OrderDto from './orderDto';
import Order from './Order';

export const handler: Handler = 
  async (event: APIGatewayProxyEvent & OrderDto ): Promise<APIGatewayProxyResult> => {
  const { orderId, product, count } = JSON.parse(event.body);

  const order: Order = {
    orderId,
    orderDate: new Date().toISOString(),
    product,
    count,
  };

  const params = {
    Entries: [
      {
        Detail: JSON.stringify(order),
        DetailType: 'OrderPlaced',
        Source: 'custom-event-bus.orders',
        EventBusName: process.env.CUSTOM_BUS,
      }
    ]
  }

  try {
    const eventbridge = new AWS.EventBridge();
    const result = await eventbridge.putEvents(params, (err, data) => {
      if (err) console.log(err, err.stack);
      else console.log(data);
    }).promise();

    return {
      statusCode: 201,
      body: `event params: ${JSON.stringify(params)}`,
    };

  } catch (err) {
    console.log(err, err);
    return {
      statusCode: 500,
      body: err
    }
  }
}

File: py-custom-event-bus/src/place_order.py
import os
import json
import uuid
from datetime import datetime
import boto3

def handler(event, context):
    message_body = json.loads(event['body'])

    order = {}
    order['order_id'] = str(uuid.uuid4())
    order['order_date'] = datetime.utcnow().strftime('%B %d, %Y')
    order['product'] = message_body['product']
    order['count'] = message_body['count']

    entries = [
                {
                    'Detail': json.dumps(order),
                    'DetailType': 'OrderPlaced',
                    'Source': 'custom-event-bus.orders',
                    'EventBusName': os.environ['CUSTOM_BUS'],
                }
            ]

    client = boto3.client('events')

    response = {}
    try:
        result = client.put_events(Entries=entries)
        print(result)
    except Exception as e:
        print(e)
        response['statusCode'] = 500
        response['body'] = str(e)
    else:
        response['statusCode'] = 200
        response['body'] = json.dumps(order) 

    return response